libminifi/test/SingleProcessorTestController.h

```diff
@@ -28,16 +28,13 @@
 #include "core/Processor.h"
 
 namespace org::apache::nifi::minifi::test {
-class SingleInputTestController : public TestController {
+class SingleProcessorTestController : public TestController {
  public:
-  explicit SingleInputTestController(const std::shared_ptr<core::Processor>& processor)
+  explicit SingleProcessorTestController(const std::shared_ptr<core::Processor>& processor)
       : processor_{plan->addProcessor(processor, processor->getName())}
   {}
 
-  std::unordered_map<core::Relationship, std::vector<std::shared_ptr<core::FlowFile>>>
-  trigger(const std::string_view input_flow_file_content, std::unordered_map<std::string, std::string> input_flow_file_attributes = {}) {
-    const auto new_flow_file = createFlowFile(input_flow_file_content, std::move(input_flow_file_attributes));
-    input_->put(new_flow_file);
+  auto trigger() {
```

Comment:
I want to keep the old abstraction, which keeps the interface similar to a transform function. We can have multiple trigger overloads instead, so the transform signature can be used for transform processors and the emitter signature for source processors. I think it would be nice if calls would only have side effects inside the processor, and only if the processor itself is stateful. `enqueueFlowFile` doesn't fulfill that.

Reply:
This has been already merged. My first implementation was just what you suggested, but changed it due to @fgerlits suggestion.

https://github.com/apache/nifi-minifi-cpp/pull/1152#discussion_r823735815

I didn't really have a strong opinion on this originally.
But now that you mentioned. I do see your point. (The previous implementation reduced boilerplate on the caller side as well.)

@fgerlits any objection to change this back to two separate trigger instead of the current enqueue + trigger?

Reply:
I like functions to perform a single operation, and `onTrigger()` is not a functional interface, so I would have gone in the opposite direction, to
```c++
putInput(createFlowFile(content, attributes));
trigger();
auto results_success = getOutput(Success);
```
but I like the idea of having a functional interface, too, so yes, it's OK to change it back.

Reply:
I've changed it back to the multiple trigger overloads in https://github.com/apache/nifi-minifi-cpp/pull/1297/commits/9d43ced3e5b168d304a3d3c832314d8e3389b84d